posthog/src/main/java/com/posthog/android/PostHogFeatureFlags.java

```diff
@@ -0,0 +1,228 @@
+    private Boolean featureFlagsLoaded;
+    private final Logger logger;
+    private final Client client;
+
+    private PostHogFeatureFlags(
+            PostHog posthog,
+            Map<String, Boolean> flagCallReported,
+            Boolean reloadFeatureFlagsQueued,
+            Boolean reloadFeatureFlagsInAction,
+            Logger logger,
+            Client client
+    ) {
+        this.posthog = posthog;
+        this.flagCallReported = flagCallReported;
+        this.reloadFeatureFlagsQueued = reloadFeatureFlagsQueued || false;
+        this.reloadFeatureFlagsInAction = reloadFeatureFlagsInAction || false;
+        this.featureFlagsLoaded = false;
+        this.logger = logger;
+        this.client = client;
+    }
+
+    public List<String> getFlags() {
+        return new ArrayList<String>(this.getFlagVariants().keySet());
+    }
+
+    public ValueMap getFlagVariants() {
+        Persistence persistence = this.posthog.persistenceCache.get();
+        return persistence.enabledFeatureFlags();
+    }
+
+    public String getFeatureFlag(final @NonNull String key, final @Nullable Map<String, Object> options) {
+        if (!this.featureFlagsLoaded) {
+            throw new IllegalStateException(String.format("getFeatureFlag for key %s failed. Feature flags didn't load in time.", key));
+        }
+        String flagValue = this.getFlagVariants().getString(key);
+        if ((Boolean) options.get("send_event") && !this.flagCallReported.get(key)) {
+            this.flagCallReported.put(key, true);
+            this.posthog.capture(
+                    "$feature_flag_called",
+                    new Properties()
+                            .putValue("$feature_flag", key)
+                            .putValue("$feature_flag_response", flagValue));
+        }
+        return flagValue;
+    }
+
+    public Boolean isFeatureEnabled(final @NonNull String key, final @Nullable Map<String, Object> options) {
+        if (!this.featureFlagsLoaded) {
+            throw new IllegalStateException(String.format("isFeatureEnabled for key %s failed. Feature flags didn't load in time.", key));
+        }
+        return this.getFeatureFlag(key, options) != null;
+    }
+
+    /**
+     * Reloads feature flags asynchronously.
+     * <p>
+     * Constraints:
+     * <p>
+     * 1. Avoid parallel requests
+     * 2. Delay a few milliseconds after each reloadFeatureFlags call to batch subsequent changes together
+     * 3. Don't call this during initial load (as /decide will be called instead), see PostHog.java
+     */
+    public void reloadFeatureFlags() {
+        if (!this.reloadFeatureFlagsQueued) {
+            this.reloadFeatureFlagsQueued = true;
+            this.startReloadTimer();
+        }
+    }
+
+    private void startReloadTimer() {
+        if (this.reloadFeatureFlagsQueued && !this.reloadFeatureFlagsInAction) {
+            new Thread(() -> {
+                try {
+                    Thread.sleep(5);
```

Comment:
Is 5 seconds the right default? and should we perhaps make this a config option

Reply:
for some context, this was copied over from the implementation in posthog-js: https://github.com/PostHog/posthog-js/blob/master/src/posthog-featureflags.js#L112

Reply:
Looking at this again, I don't think this should be configurable option that is exposed to the user. We use it internally to debounce decide requests when polling feature flags. I'll abstract it to the Utils file though—

Reply:
I think I misunderstood what this does. It is just for debouncing multiple requests so 👍